pkg/controller/controller_instance_test.go

```diff
@@ -215,7 +215,63 @@ func TestReconcileInstanceNonExistentServicePlan(t *testing.T) {
 	}
 }
 
-func TestReconcileInstanceWithParameters(t *testing.T) {
+func TestReconcileInstanceWithInvalidParameters(t *testing.T) {
+	fakeKubeClient, fakeCatalogClient, fakeBrokerClient, testController, sharedInformers := newTestController(t, noFakeActions())
+
+	sharedInformers.Brokers().Informer().GetStore().Add(getTestBroker())
+	sharedInformers.ServiceClasses().Informer().GetStore().Add(getTestServiceClass())
+
+	instance := getTestInstance()
+
+	// Secret with invalid JSON
+	invalidSecret := &v1.Secret{
+		Data: map[string][]byte{
+			testSecretKey: []byte("invalid json"),
```

Comment:
This is what I was talking about above. As a user I'm not sure it makes sense to ask me to encode my data as json when I should be able to just put the data in the secret as normal name/value pairs.

On today's call it might help if we could talk about these parameter properties at a higher level - meaning show what the UX needs are and would look like - then agree on those flows first.

Off the top of my head I think I've heard, or can think, of the following cases:
- user creates a secret to hold private data needed as params to instance/binding.create() call
- user has non-secret parameters that are needed to be passed to xxx.create() - might be alone or along with private creds from previous bullet
- user wants to use the binding/creds from one service binding as input params to another xxx.create() call
- user may need to add additional parameters (beyond the previous bullet's) to the xxx.create() call
- the keys used in parameters may use chars that are not acceptable by configMaps or secrets, so the user may need to store all/some of the param data as raw json.

Does this sound right? Are there others?